crates/ruff_linter/resources/test/fixtures/pylint/bad_super_call.py

```diff
@@ -0,0 +1,59 @@
+class Animal:
+    """Animal"""
+    pass
+
+
+class Tree:
+    """Tree"""
+    pass
+
+
+class Bug:
+    """Bug"""
+    pass
+
+class CatA(Animal):
+    """CatA"""
+    def __init__(self):
+        super(Tree, self).__init__()  # [bad-super-call]
+        super(Animal, self).__init__()
+
+
+class CatB(Animal):
+    """CatB"""
+    def __init__(self):
+        super(Animal, self).__init__() # OK
+
+
+class CatC(Animal):
+    """CatC"""
+    def some_other_func(self):
+        """Some other func"""
+        super(Tree, self).__init__()
+
+
+class CatD(Animal):
+    """CatD"""
+    def __init__(self):
+        def some_nested_func(self):
+            super(Tree, self).__init__()
+
+
+class CatE(Animal):
+    """CatE"""
+    def __init__(self):
+        super(Tree, self).__init__()
+        super(Bug, self).__init__()
+        
+        
+class CatF(Animal):
+    """CatF"""
+    def __init__(self):
+        super().__init__()
+
+        def thing():
+            super(Tree, self).__init__()
+
```

Comment:
Can we add a test with a nested class to show that it does not traverse into classes

Reply:
Something I had to ask about the nested class, why should we not traverse a nested class? Pylint doesn't pick up on it does it seem like it should?

Reply:
```
class Tree:
    """something"""


class Animal:
    """animal"""


class A:
    def __init__(self):
        class B(Tree):
            def __init__(self):
                super(Tree, self).__init__()

            def ho(self):
                print("a")

        self.b = B()


if __name__ == "__main__":
    a = A()
    a.b.ho()
```
This runs fine, but if you swap out the `Tree` for the inner class `B` you get:
```
Traceback (most recent call last):
  File "/home/chris/projects/ruff/bad_super.py", line 22, in <module>
    a = A()
        ^^^
  File "/home/chris/projects/ruff/bad_super.py", line 18, in __init__
    self.b = B()
             ^^^
  File "/home/chris/projects/ruff/bad_super.py", line 13, in __init__
    super(Animal, self).__init__()
    ^^^^^^^^^^^^^^^^^^^
TypeError: super(type, obj): obj must be an instance or subtype of type
```
So it feels like we should actually pick up on this.

Reply:
It depends on how the rule is supposed to work. I would expect that the `Checker` calls the rule for every class. That means that nested classes are covered by the `Checker` and we should not traverse nested classes too avoid flagging/traversing it twice. But maybe I misread how the rule is supposed to work.

Reply:
Hm, I believe that when I call ruff on the file that it should be calling the `Checker` once on the top level class, and once on the inner class...I can check this if you give the go-ahead for the rule.

Reply:
Hm, I believe that when I call ruff on the file that it should be calling the Checker once on the top level class, and once on the inner class...I can check this if you give the go-ahead for the rule.